### PR TITLE
[Backport release-23.11] mattermost: 8.1.4 -> 8.1.7

### PR DESCRIPTION
**Pull request by mgdelacroix**
Backport of https://github.com/NixOS/nixpkgs/pull/269691 based on the fact that the update contains security fixes

### CODE REVIEW

**Review by LeSuisse — COMMENTED** [review]
Hello,

Could please you rework the PR using `git cherry-pick -x ...` with the commits of #269691 as described in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-backport-pull-requests ?

**Comment by mgdelacroix** [comment]
@LeSuisse apologies for missing the contributing guidelines. I've cherry-picked all commits but the one that merged the master branch into the PR branch to fix conflicts. Please let me know if that's ok or if there is anything I've missed

**Review by LeSuisse — APPROVED** [review]
It looks good thanks :+1: 

I'm waiting for CI checks before merging this.

Result of `nixpkgs-review pr 271489` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mattermost</li>
  </ul>
</details>

**Review by numinit — APPROVED** [review]

**Pull request merged**
